protocol/test/foundry/PipelineConvert.t.sol

```diff
@@ -0,0 +1,1461 @@
+
+        // well is initalized with 10000 beans. cap add liquidity 
+        // to reasonable amounts.
+        amount = bound(amount, 1e6, 5000e6);
+
+        (int96 stem, uint256 lpAmountOut) = depositLPAndPassGermination(amount);
+
+        mineBlockAndUpdatePumps();
+
+        // log deltaB for this well before convert
+        int256 beforeDeltaBEth = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        int256 beforeDeltaBwsteth = seasonGetters.poolDeltaBInsta(C.BEAN_WSTETH_WELL);
+
+        // uint256 beforeGrownStalk = bs.balanceOfGrownStalk(users[1], C.BEAN_ETH_WELL);
+        uint256 beforeBalanceOfStalk = bs.balanceOfStalk(users[1]);
+
+        int96[] memory stems = new int96[](1);
+        stems[0] = stem;
+
+        AdvancedFarmCall[] memory farmCalls = new AdvancedFarmCall[](1);
+        AdvancedFarmCall[] memory lpToLPFarmCalls = createLPToLPFarmCalls(lpAmountOut);
+        farmCalls[0] = lpToLPFarmCalls[0]; // Assign the first element of the returned array
+
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = lpAmountOut;
+        
+        vm.resumeGasMetering();
+        vm.prank(users[1]);
+
+        convert.pipelineConvert(
+            C.BEAN_ETH_WELL, // input token
+            stems,  // stems
+            amounts,  // amount
+            C.BEAN_WSTETH_WELL, // token out
+            farmCalls // farmData
+        );
+    
+        int256 afterDeltaBEth = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        int256 afterDeltaBwsteth = seasonGetters.poolDeltaBInsta(C.BEAN_WSTETH_WELL);
+
+        // make sure deltaB's moved in the way we expect them to
+        assertTrue(beforeDeltaBEth < afterDeltaBEth);
+        assertTrue(afterDeltaBwsteth < beforeDeltaBwsteth);
+
+        uint256 totalStalkAfter = bs.balanceOfStalk(users[1]);
+
+        // since we didn't cross peg and there was convert power, we expect full remaining grown stalk
+        // (plus a little from the convert benefit)
+        assertTrue(totalStalkAfter >= beforeBalanceOfStalk);
+    }
+
+
+    function testUpdatingOverallDeltaB(uint256 amount) public {
+        amount = bound(amount, 1e6, 5000e6);
+        depositLPAndPassGermination(amount);
+        mineBlockAndUpdatePumps();
+
+        int256 overallDeltaB = convertGetters.overallDeltaB();
+        assertTrue(overallDeltaB != 0);
+    }
+
+
+    function testDeltaBChangeBeanToLP(uint256 amount) public {
+        amount = bound(amount, 1e6, 5000e6);
+        int256 beforeDeltaB = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        
+        doBasicBeanToLP(amount, users[1]);
+
+        int256 afterDeltaB = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        assertTrue(afterDeltaB < beforeDeltaB);
+        assertTrue(beforeDeltaB - int256(amount)*2 < afterDeltaB);
+        // would be great to calcuate exactly what the new deltaB should be after convert
+    }
+
+    function testTotalStalkAmountDidNotIncrease(uint256 amount) public {
```

Comment:
a couple of things that apply on `testTotalStalkAmountDidNotIncrease`, `testUserStalkAmountDidNotIncrease`, `testUserBDVDidNotIncrease` `testConvertAgainstPegAndLoseStalk` 
1) the setup for all of these are very similar, so i'd make a function for this. 
2) These all perform a bean -> LP convert. Is there a reason why `testBasicConvertBeanToLP` does not use `beanToLpDoConvert`? 
3) We should use the proper assert function rather than a `assertTrue`. For example, in `testTotalStalkAmountDidNotIncrease`, we should use `assertLe` here. Additionally, a comment should be written for all assertions to assist in debugging. Using `assertTrue` will only notify the dev that the boolean is false, but an extra step would be needed to check why the boolean is false.

Reply:
Fixed here: https://github.com/BeanstalkFarms/Beanstalk/commit/ddf60240eef94ab63a802c23bea6fea1d780957d